iModelCore/ECDb/ECDb/ECSql/ValueCreationFuncExp.h

```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the repository root for full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+#pragma once
+
+#include "SelectStatementExp.h"
+
+BEGIN_BENTLEY_SQLITE_EC_NAMESPACE
+
+//=======================================================================================
+//! @bsiclass
+//+===============+===============+===============+===============+===============+======
+struct ValueCreationFuncExp : ValueExp
+    {
+    private:
+        size_t m_columnRefExpIndex = UNSET_CHILDINDEX;
+        size_t m_classNameExpIndex = UNSET_CHILDINDEX;
+
+    public:
+        ValueCreationFuncExp(std::unique_ptr<DerivedPropertyExp> columnRefExp, std::unique_ptr<ClassRefExp> classRefExp, Type type) : ValueExp(type)

```

Comment:
I believe this needs to go protected because we don't want to directly create ValueCreationFuncExp, no? It also does not implement all virtual members, so it would not make sense to create an object of this class. I'm surprised the compiler does not complain about this.

Reply:
changed ctor access modifier to protected